### DISCUSSION TITLE
Is there any way to call a function by it's name(string) in Nushell?

### DISCUSSION THREAD

**Discussion by LinArcX** [nushell/nushell#7826] [Q&A]
In other Programming Languages like [Python](https://stackoverflow.com/a/3071) or Java, we have a cool feature that let the programmer introspect the internals of the language and work with them(Like calling a function by its name, etc..):

I'm curious to know if is there such a similar capability in the Nushell or not. and if not, is there any plan to add it in the future?

Scenario: I have lots of bash scripts to automate my workflow using [bemenu](https://github.com/Cloudef/bemenu). Also, I want to re-write them in Nushell scripts in near future. The problem is that I should use a long list of `case` blocks to achieve my goals:

```
option=`echo \
"Brave (Browser)
Obsidian (mind_mapper, knowledge center)
Mpv (video-audio player)
Microsoft Teams
Eva (Calculator)
Notepadqq (notepad)
Btop (ResourceMonitor)" |\
  bemenu -p "(F1)Packages:" -l 10 -i --fn "FantasqueSansMono 13" --tf "#000000" --tb "#fabd2f" --nf "#fbf1c7" --nb "#32302f" --sf "#000000" --sb "#fabd2f"  --hf "#ffc978" --hb "#005577"`

case $option in
    "Brave (Browser)") /home/linarcx/brave/brave-browser & disown ;;
    "Obsidian (mind_mapper, knowledge center)") ~/Obsidian/obsidian & disown ;;
    "Mpv (video-audio player)") mpv --player-operation-mode=pseudo-gui & disown ;;
    "Microsoft Teams") ~/teams/bin/usr/bin/teams & disown ;;
    "Eva (Calculator)") alacritty -e sh -c "eva" & disown ;;
    "Notepadqq (notepad)") notepadqq & disown ;;
    "Btop (ResourceMonitor)") alacritty -e sh -c "btop" 2&>/dev/null & disown ;;
 esac
```

As you can see, this piece of code is ugly, unmaintainable, and hard to read. even converting it to Nushell scripts maybe make it more readable, but still, I should use lots of  `if{} else{}` blocks.

The big problem is that every time I want to add a new item to bemenu list, I should mess around with `case` part. I would prefer to have separate functions for each case. with this approach, if I wanted to add new items to my script in the future, just adding a new function would be sufficient.

My idea is to have a function like this in Nushell:
```
def run [function_name: string] {
  call(function_name)
}
```
And for each item entry i create separate function.

What are your ideas?

**Comment by fdncred** [comment]
I thought I answered this already somewhere, LOL. We support having a `main` def in a nushell script that gets called automatically if it exists when you execute the script.

**Reply by LinArcX** [reply]
No, it's not about main function.

It can be any function. like "foo", "bar". 

The point is that i have only a string and i want to call it as function.

**Reply by fdncred** [reply]
Right. But if you call main with a parameter, maybe those parameters can be the name of the function you want to call and you call it with `do`. I'm not sure it would work but maybe?
